lib/templates/timers.genie

```diff
@@ -346,6 +346,11 @@ timer : Ast.FunctionCallExpression = {
         optional_once optional_emphasis at_evening => new Ast.Stream.AtTimer(null, [TIME_EVENING], null, $loader.standardSchemas.attimer).toExpression();
         optional_twice optional_per_day optional_comma optional_first optional_emphasis at_morning and optional_second optional_emphasis at_evening_second => new Ast.Stream.AtTimer(null, [TIME_MORNING, TIME_EVENING], null, $loader.standardSchemas.attimer).toExpression();
         when_brushing_teeth => new Ast.Stream.AtTimer(null, [TIME_MORNING, TIME_EVENING], null, $loader.standardSchemas.attimer).toExpression();
+
+        on date:constant_Date optional_emphasis at_noon => new Ast.Stream.OnTimer([new Date(date.getFullYear(), date.getMonth(), date.getDate(), TIME_NOON.getHours(), TIME_NOON.getMinutes(), TIME_NOON.getSeconds())]).toExpression();
```

Comment:
This does not look right. TIME_NOON is an Ast.Value.Time, it does not have methods getHours/getMinutes/getSeconds. How did this compile?

Also, in the end we didn't create OnTimerStream as a class because that would be legacy syntax.
I think this should be:
```
new Ast.FunctionCallExpression(null, 'ontimer', [new Ast.InputParam(null, 'date', new Ast.Value.Array([new Ast.DateValue(...)]))], ONTIMER_SCHEMA);
```

I'm not 100% sure about the DateValue though. The specific dates used by constant_Date are placeholders, because the utterance uses a token like DATE_0 (output of the tokenizer preprocessor). If you manipulate them, they won't match with the token in the utterance and that won't be correct.

I think we might need a new ThingTalk syntax that combines a DateValue with a TimeValue, so you would write something like:
```
"on DATE_0 at noon"
ontimer(date=[new Date(DATE_0, new Time(12, 0))]) => ...
```

```
"on DATE_0 at TIME_0"
ontimer(date=[new Date(DATE_0, TIME_0)]) => ...
```

```
"on DATE_0 in the morning"
ontimer(date=[new Date(DATE_0, $time.morning)]) => ...
```
etc.

(the tokenizer already recognizes the pattern "mm/dd/yyyy at hh:ss" and turns the whole thing into a DATE_0, so this would be for cases where date and time are separated by other words that are not just "at")

Note btw that for relative dates (that can also go into constant_Date) you won't get a DateValue that contains a Date object, you would get some other kind of DateValue which you can later combine with a TimeValue.

Reply:
Oops, I'll fix that and then look into adding additional thingtalk syntax for the additional cases. Thanks!